File: agent/commands-generator/src/main/java/org/matilda/commands/di/CommandsGeneratorComponent.java
package org.matilda.commands.di;

import dagger.Component;
import org.matilda.commands.CommandsGenerator;

@Component(modules = {AnnotationProcessorModule.class, ProcessorsModule.class, PythonModule.class})
public interface CommandsGeneratorComponent {
    CommandsGenerator commandsGenerator();
}

File: agent/commands-generator/src/main/java/org/matilda/commands/di/CommandsGeneratorComponent.kt
package org.matilda.commands.di

import androidx.room.compiler.processing.ExperimentalProcessingApi
import dagger.Component
import org.matilda.commands.CommandsGenerator

@OptIn(ExperimentalProcessingApi::class)
@Component(modules = [AnnotationProcessorModule::class, ProcessorsModule::class,
    PythonModule::class, ProtobufModule::class, JavaModule::class])
interface CommandsGeneratorComponent {
    fun commandsGenerator(): CommandsGenerator
}
